pkg/generator/pool_test.go

```diff
@@ -31,6 +37,33 @@ func TestGet(t *testing.T) {
 	}
 }
 
+// TestGetNow_EmptyPool covers the basic unhappy path when the parameter
+// pool is empty and the `GetNow` function should break the execution
+// and return an appropriate error.
+func TestGetNow_EmptyPool(t *testing.T) {
+	pool, scheduler, _ := newTestPool(
+		5,
+		func(ctx context.Context) *big.Int {
+			<-ctx.Done()
+			return nil
+		},
+	)
+	defer scheduler.stop()
+
+	_, err := pool.GetNow()
+
+	expectedErr := fmt.Errorf("pool is empty")
```

Comment:
Let's define `ErrPoolEmpty` in `pool.go` and use `testutils.AssertErrorsSame` here. The test should be cleaner but more importantly, we should be able to inspect the reason of a failure of getting parameters from the pool, if needed.